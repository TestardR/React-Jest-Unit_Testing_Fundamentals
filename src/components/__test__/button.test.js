import React from 'react';
import ReactDOM from 'react-dom';
import Button from './../button.component';

import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

afterEach(cleanup)

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Button></Button>, div);
});

it('renders button correctly', () => {
  const { getByTestId } = render(<Button label="Click Me"></Button>);
  expect(getByTestId('button')).toHaveTextContent('Click Me');
});

it("it matches snapshot", () => {
    const tree = renderer.create(<Button label="save"></Button>).toJSON()
    expect(tree).toMatchSnapshot();
})