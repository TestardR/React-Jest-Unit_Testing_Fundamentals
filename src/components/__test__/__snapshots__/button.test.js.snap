// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<button
  className="button__style"
  data-testid="button"
>
  save
</button>
`;
